common/search/src/test/scala/weco/api/search/elasticsearch/ElasticsearchServiceTest.scala

```diff
@@ -82,15 +88,176 @@ class ElasticsearchServiceTest
+          foundBySearch should have size (1)
+          foundBySearch.head shouldBe thingToQueryFor
+        }
+      }
+    }
+
+    it("returns an appropriate error when the specified index does not exist") {
+      val elasticsearchService = new ElasticsearchService(elasticClient)
+      val badIndex = createIndex
+
+      val searchRequest = searchRequestForThingByName(
+        index = badIndex,
+        name = randomAlphanumeric(10)
+      )
+
+      val findBySearchFuture =
+        elasticsearchService.findBySearch[ExampleThing](searchRequest)
+
+      whenReady(findBySearchFuture) { findBySearchEither =>
+        val failedFindBySearch = findBySearchEither.left.value
+        failedFindBySearch shouldBe a[IndexNotFoundError]
+      }
+    }
+  }
+
+  describe("findByMultiSearch") {
+    it("finds documents by performing a MultiSearch") {
+      val thingsToIndex = 0.to(randomInt(5, 10)).map(_ => randomThing).toList
+
+      withExampleIndex(thingsToIndex) { index =>
+        val elasticsearchService = new ElasticsearchService(elasticClient)
+        val thingsToQueryFor = thingsToIndex.slice(0, 3)
+
+        val searchRequests = thingsToQueryFor.map { thing =>
+          searchRequestForThingByName(
+            index = index,
+            name = thing.name
+          )
+        }
+
+        val multiSearchRequest = MultiSearchRequest(searchRequests)
+
+        val findByMultiSearchFuture = elasticsearchService
+          .findByMultiSearch[ExampleThing](multiSearchRequest)
+
+        whenReady(findByMultiSearchFuture) {
+          case (errors, foundBySearch) =>
+            errors should have length (0)
+            foundBySearch.toSet shouldBe thingsToQueryFor.toSet
+        }
+      }
+    }
+
+    it(
+      "returns an appropriate error for only those queries where the specified index does not exist"
+    ) {
+      val thingsToIndex = 0.to(randomInt(5, 10)).map(_ => randomThing).toList
+
+      withExampleIndex(thingsToIndex) { index =>
+        val elasticsearchService = new ElasticsearchService(elasticClient)
+        val thingsToQueryFor = thingsToIndex.slice(0, 3)
+
+        val badIndex = Index(randomAlphanumeric(10))
+
+        val searchRequests = thingsToQueryFor.map { thing =>
+          searchRequestForThingByName(
+            index = index,
+            name = thing.name
+          )
+        } :+ searchRequestForThingByName(
+          index = badIndex,
+          name = randomAlphanumeric(10)
+        )
+
+        val multiSearchRequest = MultiSearchRequest(searchRequests)
+
+        val findByMultiSearchFuture = elasticsearchService
+          .findByMultiSearch[ExampleThing](multiSearchRequest)
+
+        whenReady(findByMultiSearchFuture) {
+          case (errors, foundBySearch) =>
+            foundBySearch.toSet shouldBe thingsToQueryFor.toSet
+
+            errors should have size (1)
+            errors.head shouldBe a[IndexNotFoundError]
+        }
+      }
+    }
```

Comment:
Do we need a test for the case where you call findMultiSearch and the data in the index doesn't match the model?

Reply:
I think that's useful for all the test cases here. I'll add that in a future PR.